solver/monomial_solve.c

```diff
@@ -194,37 +204,128 @@ int solve_mshift_oneflavour(spinor ** const P, spinor * const Q, solver_params_t
     }
   } else
 #endif // TM_USE_QPHIX
-  if( solver_params->external_inverter == NO_EXT_INV ){
-    double reached_prec = -1.0;
-    iteration_count = cg_mms_tm(P, Q, solver_params, &reached_prec);
+  if (solver_params->type == CGMMS){
```

Comment:
Sure, I understood the purpose of that.
But in solve_degenerate and solve_mms_nd wasn't there, so I thought also here could be fine without.
Shall I add it everywhere then? If we keep as a rule that the external inverter(s) are put as first then could be fine to keep the generic case as else.. It's the same for me.

Reply:
Now that I added the check `if( solver_params->external_inverter == QPHIX_INVERTER && solver_params->type != MG )` it's more complicated.. because then in case of mms we want to keep the information that `external_inverter == QPHIX_INVERTER` as well as running MG if required.
Thus if we want to keep the check explicit for `NO_EXT_INV`; then I have to use
`if( solver_params->external_inverter == NO_EXT_INV ||  solver_params->type == MG )`.
I don't really like it. The most clean option is to make  DDalphaAMG an external solver, but still it doesn't work when I want to use both QphiX and DDalphaAMG.